iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanVisitor.java

```diff
@@ -552,36 +560,47 @@ public PlanNode visitShowTimeSeries(
                 showTimeSeriesStatement.getAuthorityScope())
             .planSchemaQueryMerge(showTimeSeriesStatement.isOrderByHeat());
 
+    boolean hasOrderBy = showTimeSeriesStatement.hasOrderByComponent();
     // show latest timeseries
     if (showTimeSeriesStatement.isOrderByHeat()
         && null != analysis.getDataPartitionInfo()
         && !analysis.getDataPartitionInfo().getDataPartitionMap().isEmpty()) {
       PlanNode lastPlanNode =
           new LogicalPlanBuilder(analysis, context).planLast(analysis, null).getRoot();
       planBuilder = planBuilder.planSchemaQueryOrderByHeat(lastPlanNode);
+      hasOrderBy = false;
```

Comment:
Why this? Shouldn't orderByComponent be null if orderByHeat is true.

Reply:
ASTVisitor will ensure that. This is for defensive programming.